operator/operator_server.go

```diff
@@ -0,0 +1,64 @@
+package operator
+
+// since operator is a serverless component on its own, its associated server
+// is only used for hosting the /metrics, /livez and /readyz endpoints
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type OperatorServer struct {
+	Mux  *http.ServeMux
+	Port int
+}
+
+func (s *OperatorServer) RegisterLivenessHandler(handler func(http.ResponseWriter) error) error {
+	if handler == nil {
+		return errors.New("liveness handler cannot be nil")
+	}
+	s.Mux.Handle("/livez", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		_ = handler(w)
```

Comment:
Is error suppression here when calling the custom handler something that should be accompanied with a warning if it errors?